src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java

```diff
@@ -99,4 +107,95 @@ public static boolean isUuid(String value) {
     }
     return false;
   }
+
+  public static String getPatientName(String json, String type) throws JsonProcessingException {
+    ObjectMapper mapper = new ObjectMapper();
+
+    JsonNode jsonNode = mapper.readTree(json);
+    JsonNode entries = jsonNode.get("entry");
+    if (entries != null) {
+      for (JsonNode entry : entries) {
+        var resourceNode = entry.get("resource");
+        if (resourceNode != null) {
+          var resourceType = resourceNode.get("resourceType");
+          if (resourceType != null && "PATIENT".equalsIgnoreCase(resourceType.asText())) {
+            JsonNode names = resourceNode.get("name");
+            if (names != null) {
+              for (JsonNode name : names) {
+                if ("family".equalsIgnoreCase(type)) {
+                  return name.get("family").asText();
+                } else if ("given".equalsIgnoreCase(type)) {
+                  JsonNode givenNames = name.get("given");
+                  if (givenNames != null) {
+                    for (JsonNode givenName : givenNames) {
+                      return givenName.asText();
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+
+  public static List<TestCaseGroupPopulation> getTestCaseGroupPopulationsFromMeasureReport(
+      String json) throws JsonProcessingException {
```

Comment:
Wondering if we can use Hl7.FHIR.r4.models to convert TestCaseJson to a FHIR Object and fetch all necessary information without parsing through each node.

Reply:
@RohitKandimalla That's a good idea, will check the model to use, thanks a lot for the suggestion!

Reply:
Talked to @RohitKandimalla, due to the release date constraint, this change will be left to future implementation.